internal/api/v1/user/merkleroots/merkleroots_sync_test.go

```diff
@@ -0,0 +1,135 @@
+package merkleroots_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	"github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/merkleroots"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/merkleroots/merklerootstest"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/clienttest"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSyncMerkleRoots(t *testing.T) {
+	t.Run("Should properly sync database when empty", func(t *testing.T) {
+		// setup
+		httpClient := resty.New()
+		server := merklerootstest.MockMerkleRootsAPIResponseNormal()
+		defer server.Close()
+
+		xpriv, err := bip32.NewKeyFromString(clienttest.UserXPriv)
+		require.NoError(t, err)
+
+		// given
+		repo := merklerootstest.CreateRepository([]models.MerkleRoot{})
+		client, err := merkleroots.NewClient(server.URL, httpClient, xpriv)
```

Comment:
Having in mind my previous comments I think it would be better here to create a general spvwallet Client and test this method using client.merkleRootsAPI.SyncMerkleRoots(). Let me know what you think

Reply:
what is general spvwallet client ?